bee-bundle/src/bundle/outgoing_bundle_builder.rs

```diff
@@ -58,17 +74,79 @@ where
     S: OutgoingBundleBuilderStage,
 {
     // TODO TEST
-    fn calculate_hash(&self) -> TritBuf {
-        // TODO Impl
+    fn calculate_bundle_hash(&mut self) -> Result<(), OutgoingBundleBuilderError> {
         let mut sponge = E::default();
+        let mut obsolete_tag = match self.builders.0.get(0) {
+            Some(builder) => match builder.obsolete_tag.as_ref() {
+                Some(obsolete_tag) => obsolete_tag.to_inner().to_owned(),
```

Comment:
Do we really have to own here?

Reply:
This will become mutable reference, but this method still have other borrow references below.

Reply:
Yeah forgot we increment it, all good!